src/models/market_data_history.rs

```diff
@@ -0,0 +1,65 @@
+use chrono::naive::serde::ts_milliseconds;
+use chrono::NaiveDateTime;
+use serde::{Deserialize, Serialize};
+
+#[derive(Default, Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct Bar {
+    pub o: f64,
+    pub c: f64,
+    pub h: f64,
+    pub l: f64,
+    pub v: i64,
+    #[serde(with = "ts_milliseconds")]
+    pub t: NaiveDateTime,
+}
+
+#[derive(Default, Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct MarketDataHistory {
+    pub server_id: String,
+    pub symbol: String,
+    pub text: String,
+    pub price_factor: i64,
+    #[serde(with = "parse_datetime")]
+    pub start_time: NaiveDateTime,
+    #[serde(default)]
+    pub high: String,
+    #[serde(default)]
+    pub low: String,
+    pub time_period: String,
+    pub bar_length: u32,
+    pub md_availability: String,
+    pub mkt_data_delay: i64,
+    pub outside_rth: bool,
+    pub trading_day_duration: Option<i64>,
+    pub volume_factor: i64,
+    pub price_display_rule: i64,
+    pub price_display_value: String,
+    pub negative_capable: bool,
+    pub message_version: i64,
+    pub data: Vec<Bar>,
+    #[serde(default)]
+    pub points: u32,
+    pub travel_time: u32,
+}
+
+mod parse_datetime {
```

Comment:
we might look into moving this into a `utils` mod, looks familiar did you grab it from somewhere else? if so, then  we shouldn't write the same code twice, we should import it.

Reply:
I derived it from your `parse_date()` function used in `futures_contract.rs`.  They should probably be handled similarly.